packages/api/src/codegen/typescript.ts

```diff
@@ -0,0 +1,357 @@
+          parameters.push({ name: 'metadata', type: data.types.params.metadata });
+          fetchArgs.push('metadata');
+        }
+      }
+
+      let returnType: string;
+      if (data.types.responses) {
+        returnType = `Promise<${Object.values(data.types.responses).join(' | ')}>`;
+      }
+
+      // @todo add docblock tags for these params and return types
+      const method = sdk.addMethod({
+        name: operationId,
+        returnType,
+        parameters,
+        statements: `return this.core.fetch(${fetchArgs.join(', ')});`,
+      });
+
+      let docblock;
+      const summary = operation.getSummary();
+      const description = operation.getDescription();
+      if (summary || description) {
+        docblock = [
+          summary ? `${summary.replace(/\n/g, '\n * ')}` : '',
+          description ? `${description.replace(/\n/g, '\n * ')}` : '',
+        ]
+          .filter(Boolean)
+          .join('\n\n');
+
+        method.addJsDoc(docblock);
+      }
+
+      // Add a typed generic HTTP method overload for this operation.
+      if (operation.method in methodGenerics) {
+        methodGenerics[operation.method].addOverload({
+          parameters: [{ name: 'path', type: 'string' }, ...parameters],
+          returnType,
+          docs: docblock ? [docblock] : null,
+        });
+      }
+    });
+
+    // @todo should all of these isolated into their own file outside of the main sdk class file?
+    // Add all known types that we're using into the SDK.
+    Array.from(this.types.values()).forEach(exp => {
+      sdkSource.addStatements(exp);
+    });
+
+    // const result = project.emitToMemory();
+    return this.project.getSourceFiles().map(sourceFile => ({
+      [sourceFile.getBaseName()]: sourceFile.getFullText(),
+    }));
+  }
+
+  /**
+   * Convert a JSON Schema object into a readily available TypeScript type or interface along with
+   * any `$ref` pointers that are in use and turn those into TS types too.
+   *
+   * Under the hood this uses https://npm.im/json-schema-to-typescript for all composition and
+   * conversion.
+   *
+   * @param schema
+   * @param name
+   */
+  async convertJSONSchemaToTypescript(schema: JSONSchema, name: string) {
+    // Though our JSON Schema type exposes JSONSchema4, which `json-schema-to-typescript`, it won't
+    // accept our custom type.
+    const ts = await compile(schema as any, name, {
+      bannerComment: '',
+    });
+
+    let primaryType: string;
+    const tmp = this.project.createSourceFile(`${name}.types.tmp.ts`, ts);
+    const declarations = tmp.getExportedDeclarations();
+
+    Array.from(declarations.keys()).forEach(declarationName => {
+      if (!primaryType) {
+        primaryType = declarationName;
+      }
+
+      declarations.get(declarationName).forEach(declaration => {
+        this.types.set(declarationName, declaration.getText());
+      });
+    });
+
+    this.project.removeSourceFile(tmp);
```

Comment:
Since we're only temporarily using `ts-morph` to do this post-processing on the `json-schema-to-typescript` types and we don't want to put those types into their own files (we'll do that later), we're immediately removing our temporary source file from the overall autogenerated SDK source code.